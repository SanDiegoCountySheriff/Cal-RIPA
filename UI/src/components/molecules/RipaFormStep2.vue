<template>
  <v-form ref="stepForm" lazy-validation>
    <template v-if="isSchool">
      <ripa-student
        v-model="model"
        v-on="$listeners"
        :disabled="disabled"
      ></ripa-student>
    </template>

    <ripa-race
      v-model="model"
      v-on="$listeners"
      :disabled="disabled"
    ></ripa-race>

    <ripa-gender
      v-model="model"
      v-on="$listeners"
      :disabled="disabled"
    ></ripa-gender>

    <ripa-age v-model="model" v-on="$listeners" :disabled="disabled"></ripa-age>

    <ripa-limited-english
      v-model="model"
      v-on="$listeners"
      :disabled="disabled"
    ></ripa-limited-english>

    <ripa-disability
      v-model="model"
      v-on="$listeners"
      :disabled="disabled"
    ></ripa-disability>

    <v-spacer></v-spacer>

    <template v-if="!isFormValid">
      <ripa-alert alert-type="error">
        Oops, you may have missed something! Please review your selections
        above.
      </ripa-alert>
    </template>

    <div class="tw-flex tw-mt-8 tw-justify-center">
      <template v-if="backButtonVisible">
        <v-btn
          outlined
          color="primary"
          class="tw-mr-2"
          :disabled="isBackNextDisabled"
          @click="handleBack"
        >
          Back
        </v-btn>
      </template>
      <v-btn outlined color="error" class="tw-mr-2" @click="handleCancel">
        Cancel
      </v-btn>
      <v-btn color="primary" :disabled="isBackNextDisabled" @click="handleNext">
        Next
      </v-btn>
    </div>
  </v-form>
</template>

<script>
import RipaPerceivedUnhoused from '@/components/molecules/RipaPerceivedUnhoused'
import RipaAge from '@/components/molecules/RipaAge'
import RipaAlert from '@/components/atoms/RipaAlert'
import RipaDisability from '@/components/molecules/RipaDisability'
import RipaGender from '@/components/molecules/RipaGender'
import RipaLimitedEnglish from '@/components/molecules/RipaLimitedEnglish'
import RipaRace from '@/components/molecules/RipaRace'
import RipaStudent from '@/components/molecules/RipaStudent'
import RipaFormStepMixin from '@/components/mixins/RipaFormStepMixin'

export default {
  name: 'ripa-form-step2',

  mixins: [RipaFormStepMixin],

  components: {
    RipaAge,
    RipaAlert,
    RipaDisability,
    RipaGender,
    RipaLimitedEnglish,
    RipaRace,
    RipaStudent,
    RipaPerceivedUnhoused,
  },

  computed: {
    isSchool() {
      return this.model?.location?.isSchool || false
    },
  },

  props: {
    backButtonVisible: {
      type: Boolean,
      default: true,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
