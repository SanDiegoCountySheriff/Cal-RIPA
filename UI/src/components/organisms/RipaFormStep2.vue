<template>
  <v-form ref="stepForm" lazy-validation>
    <ripa-student v-model="model" toggle></ripa-student>
    <ripa-race v-model="model"></ripa-race>
    <ripa-gender v-model="model"></ripa-gender>
    <ripa-age v-model="model"></ripa-age>
    <ripa-limited-english v-model="model"></ripa-limited-english>
    <ripa-disability v-model="model"></ripa-disability>

    <v-spacer></v-spacer>

    <template v-if="!isValid">
      <v-alert type="error">
        Oops, you may have missed something! Please review your selections
        above.
      </v-alert>
    </template>

    <div class="tw-flex tw-mt-8 tw-justify-center">
      <v-btn outlined color="primary" class="tw-mr-4" @click="handleBack">
        Back
      </v-btn>
      <v-btn outlined color="error" class="tw-mr-4" @click="handleCancel">
        Cancel
      </v-btn>
      <v-btn color="primary" class="tw-mr-4" @click="handleNext"> Next </v-btn>
    </div>
  </v-form>
</template>

<script>
import RipaAge from '@/components/molecules/RipaAge'
import RipaDisability from '@/components/molecules/RipaDisability'
import RipaGender from '@/components/molecules/RipaGender'
import RipaLimitedEnglish from '@/components/molecules/RipaLimitedEnglish'
import RipaRace from '@/components/molecules/RipaRace'
import RipaStudent from '@/components/molecules/RipaStudent'
import RipaFormStepMixin from '@/components/mixins/RipaFormStepMixin'

export default {
  name: 'ripa-form-step2',

  mixins: [RipaFormStepMixin],

  components: {
    RipaAge,
    RipaDisability,
    RipaGender,
    RipaLimitedEnglish,
    RipaRace,
    RipaStudent,
  },
}
</script>
