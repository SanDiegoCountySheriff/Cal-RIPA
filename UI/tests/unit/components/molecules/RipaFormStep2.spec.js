import RipaFormStep2 from '@/components/molecules/RipaFormStep2.vue'
import { shallowMount, mount } from '@vue/test-utils'
import {
  V1_STOP,
  V2_STOP,
} from '../../constants/RipaFormContainerTestConstants'
import Vuetify from 'vuetify'

describe('Ripa Form Step 2', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return shallowMount(RipaFormStep2, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaFormStep2, {
      vuetify,
      propsData: {
        value: V1_STOP,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it.todo('should display perceived unhoused for v2 stop')

  it.todo('should not display perceived unhoused for legacy stop')
})
