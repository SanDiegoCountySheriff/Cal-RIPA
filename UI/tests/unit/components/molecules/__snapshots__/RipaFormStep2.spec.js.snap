// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Form Step 2 should match snapshot 1`] = `
<form novalidate="novalidate" class="v-form">
  <!---->
  <div class="ripa-race tw-pb-4">
    <div class="ripa-form-header">
      <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
        <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Perceived Race or Ethnicity</span></div>
      </div>
    </div>
    <div class="container">
      <div class="row no-gutters">
        <div class="col-sm-12 col-12">
          <div class="ripa-check-group">
            <div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-input--selection-controls v-input--checkbox primary--text">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-marked theme--light primary--text"></i><input aria-checked="true" id="input-10" role="checkbox" type="checkbox" value="1">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-10" class="v-label theme--light" style="left: 0px; position: relative;">Asian</label>
                </div>
              </div>
            </div>
            <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-12" role="checkbox" type="checkbox" value="2">
                    <div class="v-input--selection-controls__ripple"></div>
                  </div><label for="input-12" class="v-label theme--light" style="left: 0px; position: relative;">Black/African American</label>
                </div>
              </div>
            </div>
            <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-14" role="checkbox" type="checkbox" value="3">
                    <div class="v-input--selection-controls__ripple"></div>
                  </div><label for="input-14" class="v-label theme--light" style="left: 0px; position: relative;">Hispanic/Latino(a)</label>
                </div>
              </div>
            </div>
            <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-16" role="checkbox" type="checkbox" value="4">
                    <div class="v-input--selection-controls__ripple"></div>
                  </div><label for="input-16" class="v-label theme--light" style="left: 0px; position: relative;">Middle Eastern or South Asian</label>
                </div>
              </div>
            </div>
            <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-18" role="checkbox" type="checkbox" value="5">
                    <div class="v-input--selection-controls__ripple"></div>
                  </div><label for="input-18" class="v-label theme--light" style="left: 0px; position: relative;">Native American</label>
                </div>
              </div>
            </div>
            <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-20" role="checkbox" type="checkbox" value="6">
                    <div class="v-input--selection-controls__ripple"></div>
                  </div><label for="input-20" class="v-label theme--light" style="left: 0px; position: relative;">Pacific Islander</label>
                </div>
              </div>
            </div>
            <div class="v-input theme--light v-input--selection-controls v-input--checkbox">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-22" role="checkbox" type="checkbox" value="7">
                    <div class="v-input--selection-controls__ripple"></div>
                  </div><label for="input-22" class="v-label theme--light" style="left: 0px; position: relative;">White</label>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="ripa-gender tw-pb-4">
    <div class="ripa-form-header">
      <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
        <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Perceived Gender</span></div>
      </div>
    </div>
    <div class="container">
      <div class="row no-gutters">
        <div class="col-sm-12 col-12">
          <div class="ripa-radio-group">
            <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column">
              <div class="v-input__control">
                <div class="v-input__slot" style="height: auto;">
                  <div role="radiogroup" aria-labelledby="input-33" class="v-input--radio-group__input">
                    <div class="v-radio theme--light v-item--active">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-marked theme--light primary--text"></i><input aria-checked="true" id="input-34" role="radio" type="radio" name="radio-33" value="1">
                        <div class="v-input--selection-controls__ripple primary--text"></div>
                      </div><label for="input-34" class="v-label theme--light" style="left: 0px; position: relative;">Male</label>
                    </div>
                    <div class="v-radio theme--light">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-36" role="radio" type="radio" name="radio-33" value="2">
                        <div class="v-input--selection-controls__ripple"></div>
                      </div><label for="input-36" class="v-label theme--light" style="left: 0px; position: relative;">Female</label>
                    </div>
                    <div class="v-radio theme--light">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-38" role="radio" type="radio" name="radio-33" value="3">
                        <div class="v-input--selection-controls__ripple"></div>
                      </div><label for="input-38" class="v-label theme--light" style="left: 0px; position: relative;">Transgender Male</label>
                    </div>
                    <div class="v-radio theme--light">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-40" role="radio" type="radio" name="radio-33" value="4">
                        <div class="v-input--selection-controls__ripple"></div>
                      </div><label for="input-40" class="v-label theme--light" style="left: 0px; position: relative;">Transgender Female</label>
                    </div>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div> <button type="button" class="tw-ml-4 v-btn v-btn--text theme--light v-size--x-small primary--text"><span class="v-btn__content">Clear selection</span></button>
          </div>
        </div>
        <div class="col-sm-12 col-12">
          <div class="ripa-switch" style="max-width: 250px;">
            <div class="v-input v-input--reverse v-input--expand theme--light v-input--selection-controls v-input--switch">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-46" role="switch" type="checkbox" aria-disabled="false" value="">
                    <div class="v-input--selection-controls__ripple"></div>
                    <div class="v-input--switch__track theme--light"></div>
                    <div class="v-input--switch__thumb theme--light">
                      <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                    </div>
                  </div><label for="input-46" class="v-label theme--light" style="left: 0px; position: relative;">
                    Gender Nonconforming
                  </label>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="ripa-form-header tw-mt-8">
      <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
        <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Perceived LGBT</span></div>
      </div>
    </div>
    <div class="container">
      <div class="row no-gutters">
        <div class="col-sm-12 col-12">
          <div class="ripa-switch" style="max-width: 200px;">
            <div class="v-input v-input--reverse v-input--expand v-input--hide-details theme--light v-input--selection-controls v-input--switch">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-56" role="switch" type="checkbox" aria-disabled="false" value="">
                    <div class="v-input--selection-controls__ripple"></div>
                    <div class="v-input--switch__track theme--light"></div>
                    <div class="v-input--switch__thumb theme--light">
                      <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                    </div>
                  </div><label for="input-56" class="v-label theme--light" style="left: 0px; position: relative;">
                    Perceived as LGBT
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="ripa-age tw-pb-4">
    <div class="ripa-form-header">
      <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
        <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Perceived Age</span></div>
      </div>
    </div>
    <div class="container">
      <div class="row no-gutters">
        <div class="col-sm-12 col-12">
          <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field" min="1" max="1250">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-text-field__slot"><label for="input-65" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Perceived Age</label><input inputmode="numeric" min="0" id="input-65" type="number"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="ripa-no-english tw-pb-4">
    <div class="ripa-form-header">
      <div class="row tw-ml-1 tw-mt-1">
        <!----> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
        <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Limited or No English Fluency</span></div>
      </div>
    </div>
    <div class="container">
      <div class="row no-gutters">
        <div class="col-sm-12 col-12">
          <div class="ripa-switch" style="max-width: 300px;">
            <div class="v-input v-input--reverse v-input--expand v-input--hide-details theme--light v-input--selection-controls v-input--switch">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-73" role="switch" type="checkbox" aria-disabled="false" value="">
                    <div class="v-input--selection-controls__ripple"></div>
                    <div class="v-input--switch__track theme--light"></div>
                    <div class="v-input--switch__thumb theme--light">
                      <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                    </div>
                  </div><label for="input-73" class="v-label theme--light" style="left: 0px; position: relative;">
                    Limited or no English fluency
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <!---->
  <div class="ripa-disability tw-pb-4">
    <div class="ripa-form-header">
      <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
        <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Perceived or Known Disability</span></div>
      </div>
    </div>
    <div class="container">
      <div class="row no-gutters">
        <div class="col-sm-12 col-12">
          <div class="ripa-switch" style="max-width: 200px;">
            <div class="v-input v-input--reverse v-input--expand v-input--hide-details theme--light v-input--selection-controls v-input--switch">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-82" role="switch" type="checkbox" aria-disabled="false" value="">
                    <div class="v-input--selection-controls__ripple"></div>
                    <div class="v-input--switch__track theme--light"></div>
                    <div class="v-input--switch__thumb theme--light">
                      <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                    </div>
                  </div><label for="input-82" class="v-label theme--light" style="left: 0px; position: relative;">
                    Any Disabilities?
                  </label>
                </div>
              </div>
            </div>
          </div>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <div class="spacer"></div>
  <!---->
  <div class="tw-flex tw-mt-8 tw-justify-center"><button type="button" class="tw-mr-2 v-btn v-btn--outlined theme--light v-size--default primary--text"><span class="v-btn__content">
        Back
      </span></button> <button type="button" class="tw-mr-2 v-btn v-btn--outlined theme--light v-size--default error--text"><span class="v-btn__content">
      Cancel
    </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">
      Next
    </span></button></div>
</form>
`;
